### DISCUSSION TITLE
Is it possible to just copy a text to clipboard without involving any html elements?

### DISCUSSION THREAD

**Discussion by mbonaci** [zenorocha/clipboard.js#812] [General]
When I say "involving HTML elements" I'm referring to the first param of the constructor.
I see that the library works by creating a fake input area element and then executing `document.execCommand`, but why is it necessary to be bound to some other trigger element if I just want to place some text in the clipboard?

I think the library would benefit from having a static method `Clipboard.setClipboardText(text: string)` which does the heavy lifting around clipboard without necessitating involvement of an additional HTML element.
WDYT?

**Comment by ProjectLinde37** [comment]
I think this script does not.
I use 

`    function copyText(text) {
        const listener = function (ev) {
            ev.preventDefault();
            ev.clipboardData.setData('text/plain', text);
        };
        document.addEventListener('copy', listener);
        document.execCommand('copy');
        document.removeEventListener('copy', listener);
    }`

`    function copyRichText(text) {
        const listener = function (ev) {
            ev.preventDefault();
            ev.clipboardData.setData('text/html', text);
            ev.clipboardData.setData('text/plain', text);
        };
        document.addEventListener('copy', listener);
        document.execCommand('copy');
        document.removeEventListener('copy', listener);
    }`

**Reply by mbonaci** [reply]
I've since switched to [clipboard-copy](https://github.com/feross/clipboard-copy) which has [the API](https://github.com/feross/clipboard-copy#successpromise--copytext) I was looking for.  
I'm guessing the last sentence in the API paragraph explains why this limitation might exist is copy-to-clipboard libraries in general:
> Note: in most browsers, copying to the clipboard is only allowed if copy() is triggered in direct response to a user gesture like a 'click' or a 'keypress'.
